### PR TITLE
Add mermaid figures to /docs/contribute section

### PR DESCRIPTION
**Pull request by chrismetz09**
Addresses [Issue #28808](https://github.com/kubernetes/website/issues/28808)

Mermaid figures in this PR use native docsy Mermaid support rather than the hugo mermaid shortcode. This PR does not impact existing mermaid figures that use the shortcode.

Mermaid diagrams require subgraph direction support for optimal use of available screen space.

Changes:

- Updated config.toml per [docsy look feel doc](https://www.docsy.dev/docs/adding-content/lookandfeel/#diagrams-with-mermaid)
- Updated ../partials/head.html to replace mermaid version 8.9.1 with mermaid version 8.11.2. This is because 8.9.1 does not support [mermaid subgraph directions](https://mermaid-js.github.io/mermaid/#/flowchart?id=direction-in-subgraphs). Copied 8.11.2 mermaid.min.js file into the static/js folder. 

Figures with Mermaid version 8.11.2 that included subgraph directions rendered properly. Figures with Mermaid version 8.9.1 that included subgraph directions did NOT render properly.

Mermaid diagrams added in this PR:

[Getting started for a new contributor](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IFRCXG4gICAgc3ViZ3JhcGggdGhpcmRbT3BlbiBQUl1cbiAgICAgICAgZGlyZWN0aW9uIFRCXG4gICAgICAgIFVbIF0gLS4tXG4gICAgICAgIFFbSW1wcm92ZSBjb250ZW50XSAtLT4gTltDcmVhdGUgY29udGVudF1cbiAgICAgICAgTiAtLT4gT1tUcmFuc2xhdGUgZG9jc11cbiAgICAgICAgTyAtLT4gUFtNYW5hZ2UvcHVibGlzaCBkb2NzIHBhcnRzPGJyPm9mIEs4cyByZWxlYXNlIGN5Y2xlXVxuIFxuICAgIGVuZFxuXG4gICAgc3ViZ3JhcGggc2Vjb25kW1Jldmlld11cbiAgICBkaXJlY3Rpb24gVEJcbiAgICAgICBUWyBdIC0uLVxuICAgICAgIERbTG9vayBvdmVyIHRoZTxicj5LOHMvd2Vic2l0ZTxicj5yZXBvc2l0b3J5XSAtLT5FW0NoZWNrIG91dCB0aGU8YnI-SHVnbyBzdGF0aWMgc2l0ZTxicj5nZW5lcmF0b3JdXG4gICAgICAgRSAtLT4gRltVbmRlcnN0YW5kIGJhc2ljPGJyPkdpdEh1YiBjb21tYW5kc11cbiAgICAgICBGIC0tPiBHW1JldmlldyBvcGVuIFBSPGJyPmFuZCBjaGFuZ2UgcmV2aWV3IDxicj5wcm9jZXNzZXNdXG4gICAgZW5kXG5cbiAgICBzdWJncmFwaCBmaXJzdFtTaWduIHVwXVxuICAgICAgICBkaXJlY3Rpb24gVEJcbiAgICAgICAgU1sgXSAtLi1cbiAgICAgICAgQltTaWduIHRoZSBDTkNGPGJyPkNvbnRyaWJ1dG9yPGJyPkxpY2Vuc2UgQWdyZWVtZW50XSAtLT4gXG4gICAgICAgIENbSm9pbiBzaWctZG9jczxicj5TbGFjayBjaGFubmVsXSAtLT4gTVtBdHRlbmQgd2Vla2x5PGJyPnNpZy1kb2NzIGNhbGxzPGJyPm9yIHNsYWNrIG1lZXRpbmdzXVxuICAgIGVuZFxuICAgIFxuICAgIEEoW2ZhOmZhLXVzZXIgTmV3PGJyPkNvbnRyaWJ1dG9yXSkgLS0-IGZpcnN0XG4gICAgQSAtLT4gc2Vjb25kXG4gICAgQSAtLT4gdGhpcmRcbiAgICBBIC0tPiBIW0FzayBRdWVzdGlvbnMhISFdXG4gICAgIFxuXG5jbGFzc0RlZiBncmV5IGZpbGw6I2RkZGRkZCxzdHJva2U6I2ZmZmZmZixzdHJva2Utd2lkdGg6cHgsY29sb3I6IzAwMDAwMCwgZm9udC1zaXplOjE1cHg7XG5jbGFzc0RlZiB3aGl0ZSBmaWxsOiNmZmZmZmYsc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOnB4LGNvbG9yOiMwMDAsZm9udC13ZWlnaHQ6Ym9sZFxuY2xhc3NEZWYgc3BhY2V3aGl0ZSBmaWxsOiNmZmZmZmYsc3Ryb2tlOiNmZmYsc3Ryb2tlLXdpZHRoOjBweCxjb2xvcjojMDAwXG5jbGFzcyBBLEIsQyxELEUsRixHLEgsTSxRLE4sTyxQIGdyZXlcbmNsYXNzIFMsVCxVIHNwYWNld2hpdGVcbmNsYXNzIGZpcnN0LHNlY29uZCx0aGlyZCB3aGl0ZVxuXG5cblxuXG5cbiIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6dHJ1ZX0) in docs/contribute/_index.md
[Preparation for your first contribution](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IFRCXG4gICAgc3ViZ3JhcGggdGhpcmRbT3BlbiBQUl1cbiAgICAgICAgZGlyZWN0aW9uIFRCXG4gICAgICAgIFVbIF0gLS4tXG4gICAgICAgIFFbSW1wcm92ZSBjb250ZW50XSAtLT4gTltDcmVhdGUgY29udGVudF1cbiAgICAgICAgTiAtLT4gT1tUcmFuc2xhdGUgZG9jc11cbiAgICAgICAgTyAtLT4gUFtNYW5hZ2UvcHVibGlzaCBkb2NzIHBhcnRzPGJyPm9mIEs4cyByZWxlYXNlIGN5Y2xlXVxuIFxuICAgIGVuZFxuXG4gICAgc3ViZ3JhcGggc2Vjb25kW1Jldmlld11cbiAgICBkaXJlY3Rpb24gVEJcbiAgICAgICBUWyBdIC0uLVxuICAgICAgIERbTG9vayBvdmVyIHRoZTxicj5LOHMvd2Vic2l0ZTxicj5yZXBvc2l0b3J5XSAtLT5FW0NoZWNrIG91dCB0aGU8YnI-SHVnbyBzdGF0aWMgc2l0ZTxicj5nZW5lcmF0b3JdXG4gICAgICAgRSAtLT4gRltVbmRlcnN0YW5kIGJhc2ljPGJyPkdpdEh1YiBjb21tYW5kc11cbiAgICAgICBGIC0tPiBHW1JldmlldyBvcGVuIFBSPGJyPmFuZCBjaGFuZ2UgcmV2aWV3IDxicj5wcm9jZXNzZXNdXG4gICAgZW5kXG5cbiAgICBzdWJncmFwaCBmaXJzdFtTaWduIHVwXVxuICAgICAgICBkaXJlY3Rpb24gVEJcbiAgICAgICAgU1sgXSAtLi1cbiAgICAgICAgQltTaWduIHRoZSBDTkNGPGJyPkNvbnRyaWJ1dG9yPGJyPkxpY2Vuc2UgQWdyZWVtZW50XSAtLT4gXG4gICAgICAgIENbSm9pbiBzaWctZG9jczxicj5TbGFjayBjaGFubmVsXSAtLT4gTVtBdHRlbmQgd2Vla2x5PGJyPnNpZy1kb2NzIGNhbGxzPGJyPm9yIHNsYWNrIG1lZXRpbmdzXVxuICAgIGVuZFxuICAgIFxuICAgIEEoW2ZhOmZhLXVzZXIgTmV3PGJyPkNvbnRyaWJ1dG9yXSkgLS0-IGZpcnN0XG4gICAgQSAtLT4gc2Vjb25kXG4gICAgQSAtLT4gdGhpcmRcbiAgICBBIC0tPiBIW0FzayBRdWVzdGlvbnMhISFdXG4gICAgIFxuXG5jbGFzc0RlZiBncmV5IGZpbGw6I2RkZGRkZCxzdHJva2U6I2ZmZmZmZixzdHJva2Utd2lkdGg6cHgsY29sb3I6IzAwMDAwMCwgZm9udC1zaXplOjE1cHg7XG5jbGFzc0RlZiB3aGl0ZSBmaWxsOiNmZmZmZmYsc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOnB4LGNvbG9yOiMwMDAsZm9udC13ZWlnaHQ6Ym9sZFxuY2xhc3NEZWYgc3BhY2V3aGl0ZSBmaWxsOiNmZmZmZmYsc3Ryb2tlOiNmZmYsc3Ryb2tlLXdpZHRoOjBweCxjb2xvcjojMDAwXG5jbGFzcyBBLEIsQyxELEUsRixHLEgsTSxRLE4sTyxQIGdyZXlcbmNsYXNzIFMsVCxVIHNwYWNld2hpdGVcbmNsYXNzIGZpcnN0LHNlY29uZCx0aGlyZCB3aGl0ZVxuXG5cblxuXG5cbiIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6dHJ1ZX0) in docs/contribute/_index.md
[Contributing new content preparation](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IExSIFxuICAgIHN1YmdyYXBoIHNlY29uZFtCZWZvcmUgeW91IGJlZ2luXVxuICAgIGRpcmVjdGlvbiBUQlxuICAgIFNbIF0gLS4tXG4gICAgQVtTaWduIHRoZSBDTkNGIENMQV0gLS0-IEJbQ2hvb3NlIEdpdCBicmFuY2hdXG4gICAgQiAtLT4gQ1tPbmUgbGFuZ3VhZ2UgcGVyIFBSXVxuICAgIEMgLS0-IEZbQ2hlY2sgb3V0PGJyPmNvbnRyaWJ1dG9yIHRvb2xzXVxuICAgIGVuZFxuICAgIHN1YmdyYXBoIGZpcnN0W0NvbnRyaWJ1dGluZyBCYXNpY3NdXG4gICAgZGlyZWN0aW9uIFRCXG4gICAgICAgVFsgXSAtLi1cbiAgICAgICBEW1dyaXRlIGRvY3MgaW4gbWFya2Rvd248YnI-YW5kIGJ1aWxkIHNpdGUgd2l0aCBIdWdvXSAtLT4gRVtzb3VyY2UgaW4gR2l0SHViXVxuICAgICAgIEUgLS0-IEdbXycvY29udGVudC8uLi9kb2NzJ18gZm9sZGVyIGNvbnRhaW5zIGRvY3M8YnI-Zm9yIG11bHRpcGxlIGxhbmd1YWdlc11cbiAgICAgICBHIC0tPiBIW1JldmlldyBIdWdvIHBhZ2UgY29udGVudDxicj50eXBlcyBhbmQgc2hvcnRjb2Rlc11cbiAgICBlbmRcbiAgICBcblxuICAgIGZpcnN0IC0tLS0-IHNlY29uZFxuICAgICBcblxuY2xhc3NEZWYgZ3JleSBmaWxsOiNkZGRkZGQsc3Ryb2tlOiNmZmZmZmYsc3Ryb2tlLXdpZHRoOnB4LGNvbG9yOiMwMDAwMDAsIGZvbnQtc2l6ZToxNXB4O1xuY2xhc3NEZWYgd2hpdGUgZmlsbDojZmZmZmZmLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDpweCxjb2xvcjojMDAwLGZvbnQtd2VpZ2h0OmJvbGRcbmNsYXNzRGVmIHNwYWNld2hpdGUgZmlsbDojZmZmZmZmLHN0cm9rZTojZmZmLHN0cm9rZS13aWR0aDowcHgsY29sb3I6IzAwMFxuY2xhc3MgQSxCLEMsRCxFLEYsRyxIIGdyZXlcbmNsYXNzIFMsVCBzcGFjZXdoaXRlXG5jbGFzcyBmaXJzdCxzZWNvbmQgd2hpdGVcblxuXG5cblxuXG4iLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOnRydWV9) in docs/contribute/new-content/overview.md
[Steps for opening a PR using GitHub](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IExSXG5BKFtmYTpmYS11c2VyIE5ldzxicj5Db250cmlidXRvcl0pIC0tLSBpZDFbKEs4cy9XZWJzaXRlPGJyPkdpdEh1YildXG5zdWJncmFwaCB0YXNrc1tDaGFuZ2VzIHVzaW5nIEdpdEh1Yl1cbmRpcmVjdGlvbiBUQlxuICAgIDBbIF0gLS4tXG4gICAgMVsxLiBFZGl0IHRoaXMgcGFnZV0gLS0-IDJbMi4gVXNlIEdpdEh1YiBtYXJrZG93bjxicj5lZGl0b3IgdG8gbWFrZSBjaGFuZ2VzXVxuICAgIDIgLS0-IDNbMy4gZmlsbCBpbiBQcm9wb3NlIGZpbGUgY2hhbmdlXVxuXG5lbmRcbnN1YmdyYXBoIHRhc2tzMlsgXVxuZGlyZWN0aW9uIFRCXG40WzQuIHNlbGVjdCBQcm9wb3NlIGZpbGUgY2hhbmdlXSAtLT4gNVs1LiBzZWxlY3QgQ3JlYXRlIHB1bGwgcmVxdWVzdF0gLS0-IDZbNi4gZmlsbCBpbiBPcGVuIGEgcHVsbCByZXF1ZXN0XVxuNiAtLT4gN1s3LiBzZWxlY3QgQ3JlYXRlIHB1bGwgcmVxdWVzdF0gXG5lbmRcblxuaWQxIC0tPiB0YXNrcyAtLT4gdGFza3MyXG5cbmNsYXNzRGVmIGdyZXkgZmlsbDojZGRkZGRkLHN0cm9rZTojZmZmZmZmLHN0cm9rZS13aWR0aDpweCxjb2xvcjojMDAwMDAwLCBmb250LXNpemU6MTVweDtcbmNsYXNzRGVmIHdoaXRlIGZpbGw6I2ZmZmZmZixzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6cHgsY29sb3I6IzAwMCxmb250LXdlaWdodDpib2xkXG5jbGFzc0RlZiBrOHMgZmlsbDojMzI2Y2U1LHN0cm9rZTojZmZmLHN0cm9rZS13aWR0aDoxcHgsY29sb3I6I2ZmZjtcbmNsYXNzRGVmIHNwYWNld2hpdGUgZmlsbDojZmZmZmZmLHN0cm9rZTojZmZmLHN0cm9rZS13aWR0aDowcHgsY29sb3I6IzAwMFxuY2xhc3MgQSwxLDIsMyw0LDUsNiw3IGdyZXlcbmNsYXNzIDAgc3BhY2V3aGl0ZVxuY2xhc3MgdGFza3MsdGFza3MyIHdoaXRlXG5jbGFzcyBpZDEgazhzIiwibWVybWFpZCI6IntcbiAgXCJ0aGVtZVwiOiBcImRlZmF1bHRcIlxufSIsInVwZGF0ZUVkaXRvciI6ZmFsc2UsImF1dG9TeW5jIjp0cnVlLCJ1cGRhdGVEaWFncmFtIjp0cnVlfQ) in docs/contribute/new-content/open-a-pr.md
[Working from a local fork to make your changes](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IExSXG4xW0ZvcmsgdGhlIEs4cy93ZWJzaXRlPGJyPnJlcG9zaXRvcnldIC0tPiAyW0NyZWF0ZSBsb2NhbCBjbG9uZTxicj5hbmQgc2V0IHVwc3RyZWFtXVxuc3ViZ3JhcGggY2hhbmdlc1tZb3VyIGNoYW5nZXNdXG5kaXJlY3Rpb24gVEJcblNbIF0gLS4tXG4zW0NyZWF0ZSBhIGJyYW5jaDxicj5leGFtcGxlOiBteV9uZXdfYnJhbmNoXSAtLT4gM2FbTWFrZSBjaGFuZ2VzIHVzaW5nPGJyPnRleHQgZWRpdG9yXSAtLT4gNFtcIlByZXZpZXcgeW91ciBjaGFuZ2VzPGJyPmxvY2FsbHkgdXNpbmcgSHVnbzxicj4obG9jYWxob3N0OjEzMTMpPGJyPm9yIGJ1aWxkIHRoZSBjb250YWluZXIgaW1hZ2VcIl1cbmVuZFxuc3ViZ3JhcGggY2hhbmdlczJbQ29tbWl0IC8gUHVzaF1cbmRpcmVjdGlvbiBUQlxuVFsgXSAtLi1cbjVbQ29tbWl0IHlvdXIgY2hhbmdlc10gLS0-IDZbUHVzaCBjb21taXQgdG88YnI-b3JpZ2luL215X25ld19icmFuY2hdXG5lbmRcblxuMiAtLT4gY2hhbmdlcyAtLT4gY2hhbmdlczJcblxuY2xhc3NEZWYgZ3JleSBmaWxsOiNkZGRkZGQsc3Ryb2tlOiNmZmZmZmYsc3Ryb2tlLXdpZHRoOnB4LGNvbG9yOiMwMDAwMDAsIGZvbnQtc2l6ZToxNXB4O1xuY2xhc3NEZWYgd2hpdGUgZmlsbDojZmZmZmZmLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDpweCxjb2xvcjojMDAwLGZvbnQtd2VpZ2h0OmJvbGRcbmNsYXNzRGVmIGs4cyBmaWxsOiMzMjZjZTUsc3Ryb2tlOiNmZmYsc3Ryb2tlLXdpZHRoOjFweCxjb2xvcjojZmZmO1xuY2xhc3NEZWYgc3BhY2V3aGl0ZSBmaWxsOiNmZmZmZmYsc3Ryb2tlOiNmZmYsc3Ryb2tlLXdpZHRoOjBweCxjb2xvcjojMDAwXG5jbGFzcyAxLDIsMywzYSw0LDUsNiBncmV5XG5jbGFzcyBTLFQgc3BhY2V3aGl0ZVxuY2xhc3MgY2hhbmdlcyxjaGFuZ2VzMiB3aGl0ZVxuXG4iLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOnRydWV9) in /docs/contribute/new-content/open-a-pr.md
[Steps to open a PR from your fork to the K8s/website](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IExSXG5zdWJncmFwaCBmaXJzdFsgXVxuZGlyZWN0aW9uIFRCXG4xWzEuIEdvIHRvIEs4cy93ZWJzaXRlIHJlcG9zaXRvcnldIC0tPiAyWzIuIFNlbGVjdCBOZXcgUHVsbCBSZXF1ZXN0XVxuMiAtLT4gM1szLiBTZWxlY3QgY29tcGFyZSBhY3Jvc3MgZm9ya3NdXG4zIC0tPiA0WzQuIFNlbGVjdCB5b3VyIGZvcmsgZnJvbTxicj5oZWFkIHJlcG9zaXRvcnkgZHJvcC1kb3duIG1lbnVdXG5lbmRcbnN1YmdyYXBoIHNlY29uZCBbIF1cbmRpcmVjdGlvbiBUQlxuNVs1LiBTZWxlY3QgeW91ciBicmFuY2ggZnJvbTxicj50aGUgY29tcGFyZSBkcm9wLWRvd24gbWVudV0gLS0-IDZbNi4gU2VsZWN0IENyZWF0ZSBQdWxsIFJlcXVlc3RdXG42IC0tPiA3WzcuIEFkZCBhIGRlc2NyaXB0aW9uPGJyPnRvIHlvdXIgUFJdXG43IC0tPiA4WzguIFNlbGVjdCBDcmVhdGUgcHVsbCByZXF1ZXN0XVxuZW5kXG5cbmZpcnN0IC0tPiBzZWNvbmRcblxuY2xhc3NEZWYgZ3JleSBmaWxsOiNkZGRkZGQsc3Ryb2tlOiNmZmZmZmYsc3Ryb2tlLXdpZHRoOnB4LGNvbG9yOiMwMDAwMDAsIGZvbnQtc2l6ZToxNXB4O1xuY2xhc3NEZWYgd2hpdGUgZmlsbDojZmZmZmZmLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDpweCxjb2xvcjojMDAwLGZvbnQtd2VpZ2h0OmJvbGRcbmNsYXNzIDEsMiwzLDQsNSw2LDcsOCBncmV5XG5jbGFzcyBmaXJzdCxzZWNvbmQgd2hpdGUiLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOnRydWV9) in docs/contribute/new-content/open-a-pr.md
[Review process steps](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZmxvd2NoYXJ0IExSXG4gICAgc3ViZ3JhcGggZm91cnRoW1N0YXJ0IHJldmlld11cbiAgICBkaXJlY3Rpb24gVEJcbiAgICBTWyBdIC0uLVxuICAgIE1bYWRkIGNvbW1lbnRzXSAtLT4gTltyZXZpZXcgY2hhbmdlc11cbiAgICBOIC0tPiBPW25ldyBjb250cmlidXRvcnMgc2hvdWxkPGJyPmNob29zZSBDb21tZW50XVxuICAgIGVuZFxuICAgIHN1YmdyYXBoIHRoaXJkW1NlbGVjdCBQUl1cbiAgICBkaXJlY3Rpb24gVEJcbiAgICBUWyBdIC0uLVxuICAgIEpbcmVhZCBkZXNjcmlwdGlvbjxicj5hbmQgY29tbWVudHNdLS0-IEtbcHJldmlldyBjaGFuZ2VzIGluPGJyPk5ldGxpZnkgcHJldmlldyBidWlsZF1cbiAgICBlbmRcbiBcbiAgICBBW1JldmlldyBvcGVuIFBSIGxpc3RdLS0-IEJbRmlsdGVyIG9wZW4gUFJzPGJyPmJ5IGxhYmVsXVxuICAgIEIgLS0-IHRoaXJkIC0tPiBmb3VydGhcbiAgICAgXG5cbmNsYXNzRGVmIGdyZXkgZmlsbDojZGRkZGRkLHN0cm9rZTojZmZmZmZmLHN0cm9rZS13aWR0aDpweCxjb2xvcjojMDAwMDAwLCBmb250LXNpemU6MTVweDtcbmNsYXNzRGVmIHdoaXRlIGZpbGw6I2ZmZmZmZixzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6cHgsY29sb3I6IzAwMCxmb250LXdlaWdodDpib2xkXG5jbGFzc0RlZiBzcGFjZXdoaXRlIGZpbGw6I2ZmZmZmZixzdHJva2U6I2ZmZixzdHJva2Utd2lkdGg6MHB4LGNvbG9yOiMwMDBcbmNsYXNzIEEsQixKLEssTSxOLE8gZ3JleVxuY2xhc3MgUyxUIHNwYWNld2hpdGVcbmNsYXNzIEEsQix0aGlyZCxmb3VydGggd2hpdGVcblxuXG5cblxuXG4iLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOnRydWV9) in docs/contribute/review/reviewing-prs.md

### CODE REVIEW

**Comment by sftim** [comment]
A solid LGTM from me @chrismetz09 
/lgtm

File: config.toml [hunk 1]
```diff
@@ -123,6 +123,9 @@ id = "UA-00000000-0"
 [params]
 copyright_k8s = "The Kubernetes Authors"
 copyright_linux =  "Copyright © 2020 The Linux Foundation ®."
+
+[params.mermaid]
+enable = true
```
Comment (sftim) [inline config.toml:128]: Why this change? Can you confirm?

**Comment by sftim** [comment]
Actually:
/lgtm cancel

Let's separate turning on Hugo's mermaid feature into its own PR or commit (sorry about that).

**Comment by sftim** [comment]
Adding these diagrams using the existing mechanism (the `mermaid` shortcode) would be fine. Then a separate PR can update the English docs not to use that shortcode, if that's useful.

**Comment by chrismetz09** [comment]
Tx for the review @sftim. While hacking on the diagrams, I discovered subgraph direction feature was needed. Hence the Mermaid version update. Understand I mixed a lot into this PR and per your recommendation will break it apart. 

> Adding these diagrams using the existing mechanism (the `mermaid` shortcode) would be fine. Then a separate PR can update the English docs not to use that shortcode, if that's useful.

Is this appropriate action plan?

1. Revert this Commit.
2. Revised mission for this PR: 1 commit to update Mermaid version (for subgraph direction support); another commit to add new figures (that use subgraph directions) using existing mermaid shortcode.

3. Finally new PR to transform shortcode Mermaid figures.

**Comment by sftim** [comment]
I'd split it into 2 PRs:
1. (new PR) update to a newer Mermaid version
   * there may already be a PR open for this, in which case we need to review it and merge if OK _or_ make a new one
2. (this PR) add the contribution diagram. Use same diagrams but wrap them in the mermaid shortcode.

**Comment by sftim** [comment]
@chrismetz09 would it help you if a reviewer split these changes out like I described? I'd like to see this merge _and_ I'd also like to see it merge in the right way.

**Comment by chrismetz09** [comment]
```
4197d8c32 (HEAD -> metz-pictContrib, origin/metz-pictContrib) used mermaid shortcode to render new diagrams
```
This commit modifies the new contrib figures to use the existing mermaid shortcode. The figures use a new mermaid feature for subgraph directions. Mermaid 8.9.1 implemented today does NOT support this feature. To  test new and existing mermaid figures using the shortcode with subgraph directions, the Mermaid version in this branch was updated to 8.11.2. 

A separate PR will be opened to update Mermaid version 8.9.1 to 8.11.2.

**Comment by sftim** [comment]
/retitle [WIP] added mermaid figures to /docs/contribute section

With that done, feel free to unhold @chrismetz09

**Comment by sftim** [comment]
I've just approved #29554

**Comment by sftim** [comment]
@chrismetz09 if you can, I would squash commits too.

**Comment by sftim** [comment]
Hi @chrismetz09 

What do you see as the next steps towards getting this merged?
One thing you might want to consider: narrow _this_ PR, selecting only the diagrams where you feel happy there'd be consensus about accepting them. Then you can follow it up with another PR that adds remaining diagram(s), and if we need to, we can discuss those more before we merge them.

**Comment by sftim** [comment]
If you're happy with this as-is, feel free to retitle it to remove the “[WIP]”.

**Comment by chrismetz09** [comment]
Hey @sftim, would prefer to merge this as is. Took me awhile to figure how to do first split. Rebase main done. I slacked issue with missing sidebar in local preview. Once resolved, will remove WIP, unhold it, squash commits and handle review comments.

**Comment by chrismetz09** [comment]
/retitle Add mermaid figures to /docs/contribute section

**Comment by chrismetz09** [comment]
files changed:

content/en/docs/contribute/new-content/overview.md
content/en/docs/contribute/_index.md
content/en/docs/contribute/new-content/open-a-pr.md
content/en/docs/contribute/review/reviewing-prs.md

accidental linespace in config.toml

**Comment by sftim** [comment]
nit: we might want to let folks know they can also preview locally

Thanks @chrismetz09 

IMO, just having diagrams at all makes the process look friendlier.

**Comment by chrismetz09** [comment]
> nit: we might want to let folks know they can also preview locally

Any action on my part needed? Local preview step is shown in figure at https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/new-content/open-a-pr/#fork-the-repo. Let me know and I can modify and push new commit. Tx for looking it over @sftim!

**Comment by sftim** [comment]
/lgtm

Let's get this in

**Comment by chrismetz09** [comment]
Tx for all of the guidance @sftim.

**Comment by kbhawkey** [comment]
👀

File: content/en/docs/contribute/_index.md [hunk 2]
```diff
@@ -31,6 +31,10 @@ Kubernetes documentation contributors:
 
 <!-- body -->
 
+<br>
+
+---
```
Comment (kbhawkey) [inline content/en/docs/contribute/_index.md:36]: Hi @chrismetz09
Can you check on lines 34-36? Why are these lines included on the page?

Comment (chrismetz09) [reply content/en/docs/contribute/_index.md:36]: Hey @kbhawkey, I used for line break and horizontal line between the sections on this page. I figured it tightened appearance and makes for easier reading.

Comment (kbhawkey) [reply content/en/docs/contribute/_index.md:36]: Thanks for the explanation!
I suggest to remove the horizontal line between sections. The doc pages do not use these element between headings.
I like the diagrams. Is it possible to increase the font size in the first diagram? I find the text hard to read. Thanks.

Comment (chrismetz09) [reply content/en/docs/contribute/_index.md:36]: will remove "---". I Re: font-size I tested increases from font-size:15px. We  can go to 16px before text is cropped inside. This is an isssue with Mermaid and recommended fixes involve possibly changing  CSS. I would prefer to open separate PR to handle that. Would that be okay? I want to get this in so we can point to docs url in kubecon sig docs deck. Would that be okay? Tx @kbhawkey for the review.

Comment (kbhawkey) [reply content/en/docs/contribute/_index.md:36]: That sounds okay @chrismetz09. Thanks.

**Comment by sftim** [comment]
/hold
(@chrismetz09 this is to avoid a too-early merge, but feel free to unhold when you're happy this is right)
/approve

Will still need LGTM

**Comment by kbhawkey** [comment]
Page previews:
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/review/reviewing-prs/
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/new-content/overview/
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/new-content/open-a-pr/

**Comment by chrismetz09** [comment]
Ran thru [merge conflicts doc](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#merge-conflicts-and-rebasing) and saw no conflicts. `git status` output shows unstaged `themes/docsy`. Do I have to commit that? Otherwise look good.

**Comment by sftim** [comment]
@chrismetz09 you need to rebase this against the _main_ branch in the “upstream” remote, not your local _main_ branch.

For example, this would work:
```bash
git checkout metz-pictContrib
git remote update upstream # you might use a different name than "upstream"
git rebase upstream/main
```

and this _should_ work too:
```bash
git stash push --include-untracked -- . # if you have changes pending
git checkout main
git branch -u upstream/main # you might use a different name than "upstream"
git pull --ff-only
git checkout metz-pictContrib
git rebase main
```

In both cases, if there are still conflicts, you must fix them, `git add` the fixed files, and then `git rebase --continue`.

**Comment by chrismetz09** [comment]
Tx @sftim. Learning, bare with me :-). I was attempting to follow merge docs instructions. Prior to your note, I am checked out to pict-Contrib branch. I did not run `git remote update upstream`.  But did run `git rebase upstream/main` and here is the response:

```
website git:(metz-pictContrib) ✗ git rebase upstream/main
First, rewinding head to replay your work on top of it...
Applying: Add mermaid figures to /docs/contribute section
Using index info to reconstruct a base tree...
A	content/en/docs/contribute/new-content/overview.md
.git/rebase-apply/patch:63: trailing whitespace.
    B[Sign the CNCF<br>Contributor<br>License Agreement] --- C[Join sig-docs<br>Slack channel]
.git/rebase-apply/patch:91: trailing whitespace.
You can prepare for your first contribution by reviewing several steps beforehand. The figure below outlines the steps and the details follow.
.git/rebase-apply/patch:110: trailing whitespace.

.git/rebase-apply/patch:113: trailing whitespace.

.git/rebase-apply/patch:133: trailing whitespace.
- Join the SIG Docs async Slack standup meeting on those weeks when the in-person Zoom video meeting does not take place. Meetings are always announced on `#sig-docs`. You can contribute to any one of the threads up to 24 hours after meeting announcement.
warning: squelched 8 whitespace errors
warning: 13 lines add whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging content/en/docs/contribute/new-content/_index.md
```
I see the conflict and will run your commands and report back.

**Comment by chrismetz09** [comment]
Page Previews:
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/new-content/
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/new-content/open-a-pr/
https://deploy-preview-29466--kubernetes-io-main-staging.netlify.app/docs/contribute/review/reviewing-prs/
